manifests/server_instance.pp

```diff
@@ -0,0 +1,136 @@
+# @param instance_user The user to run the instance as. Defaults to $instance_name if not specified.
+# @param instance_group The group to run the instance as. Defaults to $instance_name if not specified.
+# @param manage_instance_user_and_group Should Puppet manage the instance user and it's primary group?.
+# @param instance_directories directories needed for the instance.
+# @param initdb_settings Specifies a hash witn parameters for postgresql::server::instance::initdb
+# @param config_settings Specifies a hash with parameters for postgresql::server::instance::config
+# @param service_settings Specifies a hash with parameters for postgresql::server:::instance::service
+# @param passwd_settings Specifies a hash with parameters for postgresql::server::instance::passwd
+# @param roles Specifies a hash from which to generate postgresql::server::role resources.
+# @param config_entries Specifies a hash from which to generate postgresql::server::config_entry resources.
+# @param pg_hba_rules Specifies a hash from which to generate postgresql::server::pg_hba_rule resources.
+# @param databases Specifies a hash from which to generate postgresql::server::database resources.
+# @param databases_and_users Specifies a hash from which to generate postgresql::server::db resources.
+# @param database_grants Specifies a hash from which to generate postgresql::server::database_grant resources.
+# @param table_grants Specifies a hash from which to generate postgresql::server::table_grant resources.
+define postgresql::server_instance (
+  String[1] $instance_name                = $name,
+  Boolean $manage_instance_user_and_group = true,
+  Array $instance_directories             = [],
+  Optional[String[1]] $instance_user      = undef,
+  Optional[String[1]] $instance_group     = undef,
+  Hash $initdb_settings                   = {},
+  Hash $config_settings                   = {},
+  Hash $service_settings                  = {},
+  Hash $passwd_settings                   = {},
+  Hash $roles                             = {},
+  Hash $config_entries                    = {},
+  Hash $pg_hba_rules                      = {},
+  Hash $databases_and_users               = {},
+  Hash $databases                         = {},
+  Hash $database_grants                   = {},
+  Hash $table_grants                      = {},
+) {
+  unless($facts['os']['family'] == 'RedHat' and $facts['os']['major'] == '8') {
+    warning('This define postgresql::server_instance is only tested on RHEL8')
+  }
+  if $instance_user {
+    $real_instance_user = $instance_user
+  } else {
+    $real_instance_user = $instance_name
+  }
+  if $instance_group {
+    $real_instance_group = $instance_group
+  } else {
+    $real_instance_group = $instance_name
+  }
+
+  file { $instance_directories:
+    ensure => directory,
+  }
+
+  if $manage_instance_user_and_group {
+    User { $instance_user:
+      managehome => true,
+      home       => "/opt/pgsql/data/home/${instance_user}",
```

Comment:
I think we should make the home path configureable with a default of `"/opt/pgsql/data/home/${instance_user}"`